### PR TITLE
PPLUS-1494: Evaluator. New exceptions in private API.

### PR DESCRIPTION
**Pull request by VladislavStrelchenko**
Ticket: https://spryker.atlassian.net/browse/PPLUS-1494

ChangeLog: 
* Added new private API exceptions

### CODE REVIEW

File: src/CodeCompliance/Domain/Checks/PrivateApi/Used/AbstractUsedCodeComplianceCheck.php [hunk 1]
```diff
@@ -142,19 +142,6 @@ protected function getMethodBody(ReflectionMethod $method): ?string
...
-     *
-     * @return array<string>
-     */
-    protected function filterCoreClasses(array $coreNamespaces, array $classNamespaceList): array
-    {
-        return array_filter($classNamespaceList, function ($classNamespace) use ($coreNamespaces) {
-            return $this->hasCoreNamespace($coreNamespaces, $classNamespace);
-        });
-    }
-
```
Comment (kalaputsik) [inline src/CodeCompliance/Domain/Checks/PrivateApi/Used/AbstractUsedCodeComplianceCheck.php:157]: Please, delete 
```
//            if (substr($namespace, 0, 1) === '\\') {
//                $results[] = substr($namespace, 1);
//
//                continue;
//            }
```

Comment (kalaputsik) [inline src/CodeCompliance/Domain/Checks/PrivateApi/Used/AbstractUsedCodeComplianceCheck.php:157]: public method getParamNamespacesFromDocComment should be placed above protected

File: src/CodeCompliance/Domain/Checks/Filters/AbstractIgnoreListFilter.php [hunk 2]
```diff
@@ -50,6 +50,9 @@ public function getIgnoreSuffixes(): array
             '/\/Development\//',
             '/^Spryker\/Zed\/\w+DataImport\//',
             '/^Spryker\/Shared\/Twig\/TwigFunctionProvider$/',
+            '/^Spryker\/Zed\/Gui\/Communication\/Table\/AbstractTable$/',
+            '/^Spryker\/Zed\/Gui\/Communication\/Tabs\/AbstractTabs$/',
+            '/^Spryker\/DecimalObject\//',
```
Comment (kalaputsik) [inline src/CodeCompliance/Domain/Checks/Filters/AbstractIgnoreListFilter.php:55]: public methods should be placed above protected ones

**Review by kalaputsik — CHANGES_REQUESTED** [review]

File: src/CodeCompliance/Domain/Checks/PrivateApi/Used/AbstractUsedCodeComplianceCheck.php (hunk 1 above)
Comment (VladislavStrelchenko) [reply src/CodeCompliance/Domain/Checks/PrivateApi/Used/AbstractUsedCodeComplianceCheck.php:157]: ++

Comment (VladislavStrelchenko) [reply src/CodeCompliance/Domain/Checks/PrivateApi/Used/AbstractUsedCodeComplianceCheck.php:157]: Thanks ++

File: src/CodeCompliance/Domain/Checks/Filters/AbstractIgnoreListFilter.php (hunk 2 above)
Comment (VladislavStrelchenko) [reply src/CodeCompliance/Domain/Checks/Filters/AbstractIgnoreListFilter.php:55]: The method was changed to protected

**Review by DmytroKlymanSpryker — APPROVED** [review]

**Review by kalaputsik — APPROVED** [review]

**Review by alexanderM91 — APPROVED** [review]

**Review by DmytroKlymanSpryker — APPROVED** [review]

**Pull request merged**